### DISCUSSION TITLE
How to attach volume to container created in Go API

### DISCUSSION THREAD

**Discussion by Azer0s** [moby/moby#48660] [Q&A]
For some reason `container.Config.Volumes` is a `map[string]struct{}` and I can`t figure out how to attach a host path.

**Comment by cpuguy83** [comment]
There are 2 ways to accomplish this:

in `container.HostConfig.Binds` you have a `[]string` which takes values of `<host path>:<container path>`.
This will automatically create the host path (as a directory) if it does not exist.

In `container.HostConfig.Mounts` you have a `[]Mount` which has a `BindOptions` field you'd need to set to non-nil. This has more granular control.

https://docs.docker.com/reference/api/engine/version/v1.47/#tag/Container/operation/ContainerCreate

**Reply by Azer0s** [reply]
Okay interesting, is there any way to programatically mount NFS shares?

**Reply by cpuguy83** [reply]
Here's an example from the CLI, this should map cleanly to the mounts API.
https://stackoverflow.com/questions/39922161/mounting-nfs-shares-inside-docker-container

It the local volume driver which lets you pass arbitrary mount options to it (though its lower level than the common `mount` CLI tool).
